### ISSUE TITLE
Type errors

### ISSUE THREAD

**Issue by MrLightful** [it-gorillaz/configify#66]
<details>
<summary>Error logs</summary>

```
[psp] ERROR in ./node_modules/.pnpm/@itgorillaz+configify@1.3.1_@aws-sdk+client-secrets-manager@3.699.0_@aws-sdk+client-ssm@3.699_tstubmi6rdfjgidvudi7us7upy/node_modules/@itgorillaz/configify/src/configify.module.ts:178:20
[psp] TS2345: Argument of type 'any' is not assignable to parameter of type 'never'.
[psp]     176 |       }
[psp]     177 |
[psp]   > 178 |       exports.push(ConfigType);
[psp]         |                    ^^^^^^^^^^
[psp]     179 |       providers.push({ provide: ConfigType, useValue: instance });
[psp]     180 |     }
[psp]     181 |
[psp] 
[psp] ERROR in ./node_modules/.pnpm/@itgorillaz+configify@1.3.1_@aws-sdk+client-secrets-manager@3.699.0_@aws-sdk+client-ssm@3.699_tstubmi6rdfjgidvudi7us7upy/node_modules/@itgorillaz/configify/src/configuration/resolvers/aws/parameter-store-configuration.resolver.ts:111:21
[psp] TS2345: Argument of type 'Promise<ResolvedValue>' is not assignable to parameter of type 'never'.
[psp]     109 |     const promises = [];
[psp]     110 |     for (const [key, value] of Object.entries(config)) {
[psp]   > 111 |       promises.push(this.resolveSecretValue(key, value));
[psp]         |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[psp]     112 |     }
[psp]     113 |     return promises;
[psp]     114 |   }
[psp] 
[psp] ERROR in ./node_modules/.pnpm/@itgorillaz+configify@1.3.1_@aws-sdk+client-secrets-manager@3.699.0_@aws-sdk+client-ssm@3.699_tstubmi6rdfjgidvudi7us7upy/node_modules/@itgorillaz/configify/src/configuration/resolvers/aws/secrets-manager-configuration.resolver.ts:113:21
[psp] TS2345: Argument of type 'Promise<ResolvedValue>' is not assignable to parameter of type 'never'.
[psp]     111 |     const promises = [];
[psp]     112 |     for (const [key, value] of Object.entries(config)) {
[psp]   > 113 |       promises.push(this.resolveSecretValue(key, value));
[psp]         |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[psp]     114 |     }
[psp]     115 |     return promises;
[psp]     116 |   }
[psp] 
[psp] Found 3 errors in 3211 ms.
[psp] pnpm run start:dev:psp exited with code 0
``` 

</details>

<details>
<summary>tsconfig.json</summary>

```
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "ES2021",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true,
    "strictNullChecks": true,
    "noImplicitAny": false,
    "strictBindCallApply": false,
    "forceConsistentCasingInFileNames": false,
    "noFallthroughCasesInSwitch": false,
    "esModuleInterop": true,
    "paths": {
      ....
    }
  }
}
``` 

</details>

<details>
<summary>tsconfig.app.json</summary>

```
{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": false,
    "outDir": "../../dist/apps/psp"
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules", "dist", "test", "**/*spec.ts"]
}

``` 

</details>

Tried setting `strictPropertyInitialization` to `false`, but didn't help.

**Comment by tommelo** [comment]
@MrLightful apparently, setting `noImplicitAny` to `false` is causing this error. 
Are you able to reproduce the same error when setting `noImplicitAny` to `true`?

I will further investigate this issue, but for now, setting `"noImplicitAny": true` on your tsconfig may work.

**Comment by tommelo** [comment]
@MrLightful released a fix on `v1.3.2`. 
I was able to run it with `noImplicitAny` set to both `true` or `false`.
